From 3518f3741ef5202c894f69ca8472c644635e6277 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Thu, 6 Jun 2024 15:45:48 +0200
Subject: [PATCH] remove_host_deps_from_run

---
 pyproject.toml | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index e470755..93d2eb7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -33,9 +33,6 @@ classifiers = [
         'License :: OSI Approved :: MIT License'
 ]
 dependencies = [
-  'pybind11>=2.10.0',
-  'setuptools>=64.0.0',
-  'setuptools_scm>=8.0.0',
   'numpy>=1.14'
 ]
 
-- 
2.39.1

